.github/workflows/algolia/src/index.ts

```diff
@@ -115,15 +78,36 @@ export const generateAlgoliaJson = () => {
   return jsonData;
 };
 
-export const uploadAlgoliaData = async (records) => {
+const uploadAlgoliaData = async (index: SearchIndex, records) => {
+  return await index.saveObjects(records).catch(console.error);
+};
+
+export const syncAlgoliaIndex = async () => {
   const client = algoliasearch(
     process.env.ALGOLIA_PROJECT,
     process.env.AGOLIA_WRITE_KEY
   );
 
-  const index = client.initIndex("hash_learn");
+  const index = client.initIndex("hash_learn_testing1");
+
+  let hits: Array<{ objectID: string }> = [];
 
-  return await index
-    .saveObjects(records, { autoGenerateObjectIDIfNotExist: true })
+  await index
+    .browseObjects({
+      query: "", // Empty query will match all records
+      filters: "type:docs OR type:glossary",
+      attributesToRetrieve: ["objectID"],
+      batch: (batch) => {
+        hits = hits.concat(batch);
+      },
+    })
     .catch(console.error);
+
+  const objectIds = hits.map((hit) => hit.objectID);
+
+  index.deleteObjects(objectIds);
+
+  const records = generateAlgoliaJson();
+
+  return uploadAlgoliaData(index, records);
```

Comment:
Because `uploadAlgoliaData` is quite small and is not called from anywhere else, we can potentially flatten the logic and call `saveObjects` here. The overall flow will be easier to read.